basics_of_sst-core/Example02/ExampleEvent.h

```diff
@@ -0,0 +1,55 @@
+#ifndef _EXAMPLE_EVENT_H
+#define _EXAMPLE_EVENT_H
+
+// Basic include files from SST
+//
+#include <sst/core/event.h>
+
+namespace Example02
+{
+    // Class must inherit from SST::Event
+    //
+    class ExampleEvent : public SST::Event
+    {
+    public:
+        // Constructor
+        //
+        ExampleEvent(const uint32_t payload_value) : 
+            SST::Event(),
+            payload_(payload_value) {}
+
+        // Return the payload
+        //
+        uint32_t getPayload() const { return payload_; }
+
+    private:
+
+        // Private variables.
+        //
+        uint32_t payload_;
+
+        // Default constructor is private.
+        // This is required for serialization to work.
+        //
+        ExampleEvent() : 
+            SST::Event() {}
+
+        // Serialization code.  Serializes data before sending it out
+        // over the link.
+        //
+        void serialize_order(SST::Core::Serialization::serializer &ser) override
+        {
+            Event::serialize_order(ser);
+            ser & payload_; // Not sure what & means in this context but based
```

Comment:
Looks like this is a note to self? The serialize_order function indicates how to serialize the class. Each class member needs to be added to those in the serializer. Complex types may require special handling (e.g., serializing struct members, handling pointers correctly, etc.), but most can just be added this way.

Reply:
In this case, the "and" operator (operator&) function has been overloaded.  It is an overloaded function because there are three types of serializers: sizer, packer and unpacker.  Each one does something different.  It is specifically the & operator because this is more concise than calling a function on the serializer (i.e. ser.serialize(payload_)) and it has been used in other similar well known code (in this case boost serialization).